pkgs/development/python-modules/playwright/default.nix

```diff
@@ -0,0 +1,112 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, stdenv
+, fetchzip
+, greenlet
+, websockets
+, pyee
+, pillow
+, pixelmatch
+, autobahn
+, pyopenssl
+, service-identity
+, requests
+, objgraph
+}:
+
+buildPythonPackage rec {
+  pname = "playwright";
+  version = "1.21.0";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "playwright-python";
+    rev = "v${version}";
+    sha256 = "sha256-+q95w1UMCEZeYeXvlLagixUAROakdJFrIBy1rPMyYZc=";
+  };
+
+  format = "pyproject";
+
+  playwright-driver = stdenv.mkDerivation {
+      name = "playwright-driver";
+      src =
+        let
+        inherit (stdenv.hostPlatform) system;
+        selectSystem = attrs: attrs.${system} or (throw "Unsupported system: ${system}");
+        suffix = selectSystem {
+            x86_64-linux = "linux";
+            aarch64-linux = "linux-arm64";
+            x86_64-darwin = "mac";
+            aarch64-darwin = "mac-arm64";
+        };
+        sha256 = selectSystem {
+            x86_64-linux = "sha256-jHsxZMch8FNRgMvoLNJEgRWCbufe5SgD6WX6CPq9wgg=";
+            aarch64-linux = "sha256-XHiKYQHVcRPWa2fIfBr7k8Vu6vFz2x4N9j/pU4Y3g0s=";
+            x86_64-darwin = "sha256-SbyAsIO2lUiS8HUu6ddVIMIi6E0mm3w9SzInaGyH4Ws=";
+            aarch64-darwin = "sha256-SeEhB4cq4vo6sT1DiuBj339HGKkMK8erzY3vIjs95oQ=";
+        };
+        in
+        fetchzip {
+            url = "https://playwright.azureedge.net/builds/driver/playwright-${version}-${suffix}.zip";
+            inherit sha256;
+            postFetch = ''
+                mkdir -p $out/driver
+                install $downloadedFile $out/driver/playwright-${version}-${suffix}.zip
+            '';
+        };
+        installPhase = ''
+            mkdir -p $out/driver
+            cp -r driver/* $out/driver
+        '';
+  };
+
+  postPatch = ''
+    # works with pyee 9 as well, which is what nixos has packaged
+    # disables setuptools-scm, as it fails
+    substituteInPlace meta.yaml --replace "- pyee ==8.1.0" ""
+    substituteInPlace setup.py \
+        --replace '"pyee==8.1.0",' "" \
+        --replace '"setuptools-scm==6.3.2",' ""
+    # replace functionality that setuptools-scm provides by adding the version to a file
+    echo 'version = "${version}"' > playwright/_repo_version.py
+  '';
+
+  preBuild = ''
+    mkdir -p driver
+    cp -r ${playwright-driver}/driver ./
+  '';
+
+  preInstall = ''
+    # As pip names the wheel "dist", but when installing it will expect a properly formatted filename in dist
+    mv dist dist.orig
+    mkdir -p dist
+    mv dist.orig dist/playwright-${version}-py3-none-any.whl
+  '';
+
+  propagatedBuildInputs = [
+      playwright-driver
+      greenlet
+      websockets
+      pyee
+      pillow
+      pixelmatch
+      autobahn
+      pyopenssl
+      service-identity
+      requests
+      objgraph
+    ];
+
+  # skip tests because they require network access
+  doCheck = false;
```

Comment:
Check if you can skip only those networky tests?

Reply:
Yeah, but since pretty much all of them are networky (the point of this library is to automate browser things), `disabledTestPaths` would be looong.

As a simple smoke test I've added an import check.